### PR TITLE
fix SQS JSON protocol support in ASF

### PR DESCRIPTION
**Pull request by alexrashed**
This PR aims at providing JSON support for SQS. AWS supports both protocols, `query` and `json`, for SQS for some time.

- With version `1.27.127` of [botocore](https://github.com/boto/botocore) (the official AWS Python SDK clients) the protocol of the SQS service spec has changed from `query` to `json`: https://github.com/boto/botocore/commit/2395012bd369f89b0bcfbbf830718866fc6ae166
  - This broke the usage of SQS with the AWS Python SDK for that underlying botocore version as reported in #8267.
- With version `1.27.129` this change reverted it again: https://github.com/boto/botocore/pull/2931
- With version `1.31.81` the switch was reintroduced and the issue effects us again: https://github.com/boto/botocore/commit/84c784761d4cde0c0ce3b2b32382f970d23b3cc8
  - Since `2023-11-09`, the JSON protocol is generally available as announced in this [AWS blogpost](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-sqs-support-json-protocol/)
### Initially explored solution:
- [x] Upgrade to the latest version of `botocore` internally.
- [x] Implement a parser and serializer facade which can dynamically select the right parser/serializer based on the content type / accept headers of the request.
- [x] Implement error code handling for `x-amzn-query-error`, as implemented in https://github.com/boto/botocore/pull/2768
- [x] Upgrade to a moto which contains a fix for https://github.com/getmoto/moto/issues/6286
- [ ] ~~Implement tests which ensure that both, query-protocol and json-protocol requests and responses can be sent for SQS.~~
- [ ] ~~Fix all other tests that might break with the upgrade of botocore.~~
  - ~~Try to make SQS tests protocol agnostic, and the parser tests protocol specific (try not to depend on botocore spec protocol)~~

As it turns out this solution is not feasible, because the JSON specification removed a lot of information from the spec which is used to properly parse and serialize the requests and responses.
This means that there is no other alternative than handling two different specifications.

### New solution
- [x] Upgrade to the latest version of `botocore` internally.
- [x] Add the latest available version of the SQS query specifications to `localstack.aws.data`.
  - This is a newly created folder which is configured as `extra search path` in `botocore` when loading the specifications.
  - This will add a newly created "alias" service called `sqs-query`.

What's left to do:
- [x] Check if it is necessary to support JSON requests for SQS Queue URLs (or implement content negotiation).
  - Not necessary per se, it is not used by the `json` protocol based SDK. Would be nice to implement soon.
- [x] Check if the internal dev endpoints for SQS should be migrated to JSON (or implement content negotiation).
  - those endpoint are not targeted by SDKs, so it is not necessary right now, and can be tackled in a second iteration
- [x] Fix failing tests.
  - Most of them should only fail because they assume an XML response structure, but due to the upgrade of botocore a JSON request (and with that a JSON response) is sent.
- [x] Assert that the implementation works with some tests explicitly using XML
- [x] Ensure compatibility with ext (-ext is mostly green, only 4 tests to fix there with wrong assert)

### CODE REVIEW

**Comment by shadowhand** [comment]
As of version 3.285.2 of aws/aws-sdk (for PHP) SQS::GetQueueUrl calls to localstack fail. This means we cannot upgrade the SDK because our CI/CD servers use localstack for tests, and all builds with version 3.285.2 are failing.

**Comment by alexrashed** [comment]
Hi @shadowhand! 
I am really sorry for the inconvenience. However, as you can see here, we are already working on this issue in this PR. 
If you want to discuss the severity or other aspects of this issue, please head over to the related issue: https://github.com/localstack/localstack/issues/8267
Thanks!

File: localstack/aws/protocol/service_router.py
Comment (bentsku) [inline localstack/aws/protocol/service_router.py:360]: I have the commit ready to remove this, but the CI is green, so will push and merge straight after review is no change is necessary

File: localstack/aws/protocol/serializer.py
Comment (bentsku) [inline localstack/aws/protocol/serializer.py:1636]: will fix this typo, commit is ready

File: localstack/utils/coverage_docs.py
Comment (bentsku) [inline localstack/utils/coverage_docs.py:13]: https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-json-faqs.html#query-protocol

I don't think the SQS Query (XML) implementation is going to be phased out soon

**Review by alexrashed — COMMENTED** [review]
Thanks so much @bentsku for finalizing the PR and figuring out the nitty gritty details of the remaining test failures! 🦸🏽 
From my perspective, this PR is ready to be merged as it is, but I would like to get another pair of eyes from @dominikschubert or @thrau before merging this.
I'm not happy with some of the changes in here (like the hacky "provider alias"), but I think we can iron these things out in a follow-up.

**Review by dominikschubert — APPROVED** [review]
LGTM, Thanks so much @alexrashed  and @bentsku  for getting this over the line so quickly!

It's probably not going to be the only case where they'll make the switch to JSON I guess :disappointed: 

As @bentsku  said, any typos, etc. can be removed in a follow-up. First let's get this merged :rocket:

**Comment by ilia-beliaev-miro** [comment]
Looks like this change has broke the localstack SQS usage for us due to the service name change from `sqs` to `sqs-query`. We receive `Service 'sqs-query' is not enabled.` now, was it an expected outcome and we should now switch the service name when creating a localstack instance from `sqs` to `sqs-query`?

**Comment by bentsku** [comment]
Hello @ilia-beliaev-miro, this is an unexpected outcome and we will fix it with #9607. Sorry for the issue, you can declare both service names for now, but it should be fixed by the end of the day!

**Pull request merged**
